src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/ISubscriptionStorage.cs

```diff
@@ -17,5 +17,11 @@ public interface ISubscriptionStorage
         /// Unsubscribes the given client address from messages of the given types.
         /// </summary>
         Task Unsubscribe(string client, IEnumerable<MessageType> messageTypes, SubscriptionStorageOptions options);
+
+        /// <summary>
+        /// Returns a list of addresses of subscribers that previously requested to be notified
+        /// of messages of the given message types.
```

Comment:
Not sure yet how to call this thing. I am pretty sure it will change a bit since it makes more sense to subscribe by passing an endpoint name instead of a physical address.